editor/filesystem_dock.cpp

```diff
@@ -2141,6 +2141,135 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
+#elif defined(LINUXBSD_ENABLED)
+				// Try terminal emulators that ship with common Linux distributions first.
+				terminal_emulators.push_back("gnome-terminal");
+				terminal_emulators.push_back("konsole");
+				terminal_emulators.push_back("xfce4-terminal");
+				terminal_emulators.push_back("lxterminal");
+				terminal_emulators.push_back("kitty");
+				terminal_emulators.push_back("alacritty");
+				terminal_emulators.push_back("urxvt");
+				terminal_emulators.push_back("xterm");
+#endif
+			} else {
+				// Use the user-specified terminal.
+				terminal_emulators.push_back(terminal_emulator_setting);
+			}
+
+			String arguments = EDITOR_GET("filesystem/external_programs/terminal_emulator_flags");
+			if (arguments.is_empty()) {
+				// NOTE: This default value is ignored further below if the terminal executable is `powershell` or `cmd`,
+				// due to these terminals requiring nonstandard syntax to start in a specified folder.
+				arguments = "{directory}";
+			}
+
+#ifdef LINUXBSD_ENABLED
+			String chosen_terminal_emulator;
+			for (const String &terminal_emulator : terminal_emulators) {
+				List<String> test_args; // Required for `execute()`, as it doesn't accept `Vector<String>`.
+				test_args.push_back("-v");
+				test_args.push_back(terminal_emulator);
+				// Silence command name being printed when found. (stderr is already silenced by `OS::execute()` by default.)
+				// FIXME: This doesn't appear to silence stdout.
+				test_args.push_back(">");
+				test_args.push_back("/dev/null");
+				int exit_code = 0;
+				const Error err = OS::get_singleton()->execute("command", test_args, nullptr, &exit_code);
+				if (err == OK && exit_code == EXIT_SUCCESS) {
+					chosen_terminal_emulator = terminal_emulator;
+					break;
+				} else if (err == ERR_CANT_FORK) {
+					ERR_PRINT_ED(vformat(TTR("Couldn't run external program to check for terminal emulator presence: command -v %s"), terminal_emulator));
+				}
+			}
+#else
+			// On Windows and macOS, the first (and only) terminal emulator in the list is always available.
+			String chosen_terminal_emulator = terminal_emulators[0];
+#endif
+
+			List<String> terminal_emulator_args; // Required for `execute()`, as it doesn't accept `Vector<String>`.
+#ifdef LINUXBSD_ENABLED
+			// Prepend default arguments based on the terminal emulator name.
+			// Use `String.ends_with()` so that installations in non-default paths
+			// or `/usr/local/bin` are detected correctly.
+			if (chosen_terminal_emulator.ends_with("konsole")) {
+				terminal_emulator_args.push_back("--workdir");
+			}
+#endif
+
+			bool append_default_args = true;
+
+#ifdef WINDOWS_ENABLED
+			// Prepend default arguments based on the terminal emulator name.
+			// Use `String.get_basename().to_lower()` to handle Windows' case-insensitive paths
+			// with optional file extensions for executables in `PATH`.
+			if (chosen_terminal_emulator.get_basename().to_lower() == "powershell") {
```

Comment:
This could also be equal to `pwsh`, if the user has a newer version of Powershell (i.e. "Powershell Core" in addition to "Windows Powershell") installed. Same command line arguments would apply.

Reply:
I think Windows itself still calls `powershell` by default, but you should be able to specify `pwsh` in the custom terminal name editor setting. I can modify the class reference accordingly.